python/lsst/ts/athexapod/csc.py

```diff
@@ -186,129 +176,53 @@ async def configure(self, config):
-            # TODO Add make_ackcmd call when it is released.
-        except Exception as e:
-            self.log.exception(e)
-            raise e
-
-        self.run_telemetry_task = True
-        self.telemetry_task = asyncio.create_task(self.telemetry())
-
-        await super().end_start(data)
-
-    async def end_standby(self, data):
-        """Executes after transition from DISABLE to STANDBY.
-
-        It will stop the telemetry loop and disconnect from the hexapod
-        controller.
-        """
-
-        try:
-            await self.close_telemetry_task()
-        except Exception:
-            self.log.exception("Exception closing telemetry task.")
-
-        try:
-            await self.controller.disconnect()
-        except Exception:
-            self.log.exception("Exception disconnecting from hexapod controller.")
-
-        await super().end_standby(data)
-
-    async def end_enable(self, data):
-        """Executed after state is enabled.
-
-        This may cause the hexapod to move.
-
-        Actions performed by this task are:
-
-        1. - Set soft limits on the hexapod controller.
-        2. - Check that the hexapod axis are referenced.
-        3. - If axis are not referenced and `auto_reference=True`, will
-             reference axis.
-
-        The CSC will reject attempts to move axis that are not referenced.
-
-        Parameters
-        ----------
-        data
-        """
-        await self.report_new_ready(False)
-        await self.report_detailed_state(ATHexapod.DetailedState.NOTINMOTION)
-
-        # Set soft limits
-
-        await self.set_limits(
-            self.config.limit_xy_max,
-            self.config.limit_z_min,
-            self.config.limit_z_max,
-            self.config.limit_uv_max,
-            self.config.limit_w_min,
-            self.config.limit_w_max,
-        )
-
-        try:
-            await self.assert_ready("enable")
-        except salobj.base.ExpectedError:
-            self.log.exception("Hexapod controller not ready.")
-            await self.fault(
-                code=CONTROLLER_NOT_READY,
-                report="Hexapod controller not ready.",
-                traceback=traceback.format_exc(),
-            )
-
-        ref = await self.is_referenced()
-
-        if not ref:
-            self.log.warning("Referencing Axis.")
-            await self.controller.reference()
-            await self.report_detailed_state(ATHexapod.DetailedState.INMOTION)
-            try:
-                await asyncio.wait_for(
-                    self.wait_movement_done(), timeout=self.config.reference_timeout
-                )
-            except asyncio.TimeoutError as e:
-                self.log.error("Referencing time out.")
-                self.log.exception(e)
-                await self.fault(
-                    code=REFERENCING_TIMEOUT,
-                    report="Referencing time out.",
-                    traceback=traceback.format_exc(),
```

Comment:
I am not happy with any of these changes. This class was designed like that with a reason and changing this without further explanation will not do. Please revert.